internal/platform/storage/couchbase.go

```diff
@@ -0,0 +1,258 @@
+		err = cluster.Manager(username, password).InsertBucket(&gocb.BucketSettings{
+			Name:  tokensBucketName,
+			Quota: defaultBucketQuota,
+		})
+		if err != nil {
+			return nil, errors.Wrap(err, "can't create 'tokens' bucket")
+		}
+
+		// TODO figure out better way how to wait until bucket will be ready
+		// see also https://forums.couchbase.com/t/bucket-creation-callback/17220
+		time.Sleep(2 * time.Second)
+
+		tokens, err := cluster.OpenBucket(tokensBucketName, bucketPassword)
+		if err != nil {
+			return nil, errors.Wrap(err, "can't open 'tokens' bucket")
+		}
+
+		if err := tokens.Manager(username, password).CreatePrimaryIndex("", true, false); err != nil {
+			return nil, errors.Wrap(err, "can't create primary index in 'tokens' bucket")
+		}
+	}
+
+	return &Storage{cluster: cluster}, nil
+}
+
+func (s *Storage) CreateUser(id uuid.UUID, email, hashedPassword string) error {
+	user := User{
+		ID:       id,
+		Email:    email,
+		Password: string(hashedPassword),
+	}
+
+	users, err := s.cluster.OpenBucket("users", "")
+	if err != nil {
+		return errors.Wrap(err, "failed to open bucket")
+	}
+
+	defer users.Close()
+
+	_, err = users.Upsert(id.String(), user, 0)
+	if err != nil {
+		return errors.Wrap(err, "failed to upsert user")
+	}
+
+	return nil
+}
+
+func (s *Storage) GetUserByID(id string) (User, error) {
+	var user User
+
+	query := gocb.NewN1qlQuery("select Email from users where ID=$userID")
+
+	params := make(map[string]interface{})
+	params["userID"] = id
+
+	users, err := s.cluster.OpenBucket("users", "")
+	if err != nil {
+		return user, errors.Wrap(err, "failed to open bucket")
+	}
+
+	defer users.Close()
+
+	rows, err := users.ExecuteN1qlQuery(query, params)
+	if err != nil {
+		return user, errors.Wrap(err, "can't get user")
+	}
+
+	rows.Close()
+
+	if rows.Metrics().ResultCount == 0 {
+		return user, ErrUserNotFound
+	}
+
+	for rows.Next(&user) {
+		break
+	}
+
+	return user, nil
+}
+
+func (s *Storage) GetUserByEmail(email string) (User, error) {
+	var user User
+
+	query := gocb.NewN1qlQuery("select `ID`,`Email`,`Password` from `users` where `Email`=$email")
+
+	params := make(map[string]interface{})
+	params["email"] = email
+
+	users, err := s.cluster.OpenBucket(usersBucketName, "")
+	if err != nil {
+		return user, errors.Wrap(err, "can't open 'users' bucket")
+	}
+
+	defer users.Close()
+
+	rows, err := users.ExecuteN1qlQuery(query, params)
+	if err != nil {
+		return user, errors.Wrap(err, "can't get user")
+	}
+
+	rows.Close()
+
+	if rows.Metrics().ResultCount == 0 {
+		return User{}, ErrUserNotFound
+	}
+
+	for rows.Next(&user) {
+		break
+	}
+
+	return user, nil
+}
+
+func (s *Storage) CreateToken(token string, claims Claims) error {
+	tokenData := Token{
+		Token:  token,
+		Claims: claims,
+	}
+
+	tokens, err := s.cluster.OpenBucket(tokensBucketName, "")
+	if err != nil {
+		return errors.Wrap(err, "can't open 'tokens' bucket")
+	}
+
+	defer tokens.Close()
+
+	_, err = tokens.Upsert(token, tokenData, 0)
```

Comment:
We can use timeout for this record to remove old tokens from DB automatically. If token not exists in DB it is not valid

Reply:
Why we even store token in DB?

Reply:
I was thinking about set expiration on record. But I decided to save tokens and just modify expiration time.